src/arduino/arduino.ts

```diff
@@ -673,11 +673,18 @@ export class ArduinoApp {
         if (buildDir || dc.output) {
             // 2020-02-29, EW: This whole code appears a bit wonky to me.
             //   What if the user specifies an output directory "../builds/my project"
-            buildDir = path.resolve(ArduinoWorkspace.rootPath, buildDir || dc.output);
+
+            // the first choice of the path should be from the users explicit settings.
+            if (dc.output) { 
+                buildDir = path.resolve(ArduinoWorkspace.rootPath, dc.output);
+            } else {
+                buildDir = path.resolve(ArduinoWorkspace.rootPath, buildDir);
+            }
+
+
             const dirPath = path.dirname(buildDir);
             if (!util.directoryExistsSync(dirPath)) {
-                logger.notifyUserError("InvalidOutPutPath", new Error(constants.messages.INVALID_OUTPUT_PATH + buildDir));

```

Comment:
How was the output directory created before this change, and will the new way of creating the output directory cause any conflicts with the old code path?

Also, if the `INVALID_OUTPUT_PATH` constant is no longer used then it should be removed.

Reply:
Before this change, the output directory wasn't created. Specifically, it seemed that the only case this affected was when a user had an output set in their arduino.json upon opening VSCode, this would happen. I see no real reason why not to go ahead and create the folder at this point.